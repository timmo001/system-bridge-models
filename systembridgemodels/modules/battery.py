"""Battery."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Battery:
    """Battery."""

    is_charging: bool | None = None
    percentage: float | None = None
