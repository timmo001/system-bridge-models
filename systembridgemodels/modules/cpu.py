"""CPU."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class CPUFrequency:
    """CPU Frequency."""

    current: float | None = None
    min: float | None = None
    max: float | None = None


@dataclass
class CPUStats:
    """CPU Stats."""

    ctx_switches: int | None = None
    interrupts: int | None = None
    soft_interrupts: int | None = None
    syscalls: int | None = None


@dataclass
class CPUTimes:
    """CPU Times."""

    user: float | None = None
    system: float | None = None
    idle: float | None = None
    interrupt: float | None = None
    dpc: float | None = None


@dataclass
class CPU:
    """CPU."""

    count: int | None = None
    frequency: CPUFrequency | None = None
    frequency_per_cpu: list[CPUFrequency] | None = None
    load_average: float | None = None
    power_package: float | None = None
    power_per_cpu: list[float] | None = None
    stats: CPUStats | None = None
    temperature: float | None = None
    times: CPUTimes | None = None
    times_per_cpu: list[CPUTimes] | None = None
    times_percent: CPUTimes | None = None
    times_percent_per_cpu: list[CPUTimes] | None = None
    usage: float | None = None
    usage_per_cpu: list[float] | None = None
    voltage: float | None = None
