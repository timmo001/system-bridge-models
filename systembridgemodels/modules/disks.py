"""Disks."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class DiskIOCounters:
    """Disk IO Counters."""

    read_count: int
    write_count: int
    read_bytes: int
    write_bytes: int
    read_time: int
    write_time: int


@dataclass
class DiskUsage:
    """Disk Usage."""

    total: int
    used: int
    free: int
    percent: float


@dataclass
class DiskPartition:
    """Disk Partition."""

    device: str
    mount_point: str
    filesystem_type: str
    options: str
    max_file_size: int
    max_path_length: int
    usage: DiskUsage | None = None


@dataclass
class Disk:
    """Disk."""

    name: str
    partitions: list[DiskPartition]
    io_counters: DiskIOCounters | None = None


@dataclass
class Disks:
    """Disks."""

    devices: list[Disk]
    io_counters: DiskIOCounters | None = None
