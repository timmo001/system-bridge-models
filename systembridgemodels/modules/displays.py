"""Displays."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Display:
    """Display."""

    name: str
    resolution_horizontal: int
    resolution_vertical: int
    x: int
    y: int
    width: int | None = None
    height: int | None = None
    is_primary: bool | None = None
    pixel_clock: float | None = None
    refresh_rate: float | None = None


type Displays = list[Display]
