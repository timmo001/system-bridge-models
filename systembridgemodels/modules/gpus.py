"""GPUs."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class GPU:
    """GPU."""

    name: str
    core_clock: float | None = None
    core_load: float | None = None
    fan_speed: float | None = None
    memory_clock: float | None = None
    memory_load: float | None = None
    memory_free: float | None = None
    memory_used: float | None = None
    memory_total: float | None = None
    power_usage: float | None = None
    temperature: float | None = None


type GPUs = list[GPU]
