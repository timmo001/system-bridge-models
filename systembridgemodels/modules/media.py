"""Media."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Media:
    """Media Info."""

    album_artist: str | None = None
    album_title: str | None = None
    artist: str | None = None
    duration: float | None = None
    is_fast_forward_enabled: bool | None = None
    is_next_enabled: bool | None = None
    is_pause_enabled: bool | None = None
    is_play_enabled: bool | None = None
    is_previous_enabled: bool | None = None
    is_rewind_enabled: bool | None = None
    is_stop_enabled: bool | None = None
    playback_rate: float | None = None
    position: float | None = None
    repeat: str | None = None
    shuffle: bool | None = None
    status: str | None = None
    subtitle: str | None = None
    thumbnail: str | None = None
    title: str | None = None
    track_number: int | None = None
    type: str | None = None
    updated_at: float | None = None
