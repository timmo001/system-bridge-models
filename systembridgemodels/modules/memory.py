"""Memory."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class MemorySwap:
    """Memory Swap."""

    total: int | None = None
    used: int | None = None
    free: float | None = None
    percent: float | None = None
    sin: int | None = None
    sout: int | None = None


@dataclass
class MemoryVirtual:
    """Memory Virtual."""

    total: int | None = None
    available: int | None = None
    percent: float | None = None
    used: int | None = None
    free: int | None = None
    active: int | None = None
    inactive: int | None = None
    buffers: int | None = None
    cached: int | None = None
    wired: int | None = None
    shared: int | None = None


@dataclass
class Memory:
    """Memory."""

    swap: MemorySwap
    virtual: MemoryVirtual
