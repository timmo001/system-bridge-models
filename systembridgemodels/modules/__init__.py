"""Modules."""
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum

from .battery import Battery
from .cpu import CPU
from .disks import Disks
from .displays import Displays
from .gpus import GPUs
from .media import Media
from .memory import Memory
from .networks import Networks
from .processes import Processes
from .sensors import Sensors
from .system import System


class DataEnum(Enum):
    """Data Enum."""

    BATTERY = "battery"
    CPU = "cpu"
    DISKS = "disks"
    DISPLAYS = "displays"
    GPUS = "gpus"
    MEDIA = "media"
    MEMORY = "memory"
    NETWORKS = "networks"
    PROCESSES = "processes"
    SENSORS = "sensor"
    SYSTEM = "system"


@dataclass
class GetData:
    """Get Data."""

    modules: list[str]


@dataclass
class RegisterDataListener(GetData):
    """Register Data Listener."""


@dataclass
class ModulesData:
    """Data."""

    battery: Battery | None = None
    cpu: CPU | None = None
    disks: Disks | None = None
    displays: Displays | None = None
    gpus: GPUs | None = None
    media: Media | None = None
    memory: Memory | None = None
    networks: Networks | None = None
    processes: Processes | None = None
    sensors: Sensors | None = None
    system: System | None = None
