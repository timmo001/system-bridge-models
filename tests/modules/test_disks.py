"""Test the disks module model."""

from dataclasses import asdict

from syrupy.assertion import SnapshotAssertion

from systembridgemodels.fixtures.modules.disks import FIXTURE_DISKS
from systembridgemodels.modules.disks import (
    Disk,
    DiskIOCounters,
    DiskPartition,
    Disks,
    DiskUsage,
)


def _test_disk_partition(partition: DiskPartition):
    assert isinstance(partition, DiskPartition)
    assert partition.device == "device"
    assert partition.mount_point == "mountpoint"
    assert partition.filesystem_type == "fstype"
    assert partition.options == "options"
    assert partition.max_file_size == 1
    assert partition.max_path_length == 2
    assert isinstance(partition.usage, DiskUsage)
    assert partition.usage.total == 1
    assert partition.usage.used == 2
    assert partition.usage.free == 3
    assert partition.usage.percent == 40.2


def _test_disk(disk: Disk):
    assert isinstance(disk, Disk)
    assert disk.name == "name"
    assert isinstance(disk.partitions, list)
    assert isinstance(disk.partitions[0], DiskPartition)
    _test_disk_partition(disk.partitions[0])
    assert isinstance(disk.io_counters, DiskIOCounters)
    assert disk.io_counters.read_bytes == 1
    assert disk.io_counters.write_bytes == 2
    assert disk.io_counters.read_count == 3
    assert disk.io_counters.write_count == 4
    assert disk.io_counters.read_time == 5
    assert disk.io_counters.write_time == 6


def test_disks(snapshot: SnapshotAssertion):
    """Test the disks model."""
    disks = FIXTURE_DISKS
    assert isinstance(disks, Disks)
    assert disks == snapshot


def test_disks_dict(snapshot: SnapshotAssertion):
    """Test disks dict."""
    disks_dict = asdict(FIXTURE_DISKS)
    assert isinstance(disks_dict, dict)
    assert disks_dict == snapshot

    disks_converted = Disks(**disks_dict)
    assert isinstance(disks_converted, Disks)
    assert disks_converted == snapshot(
        name="disks-dict-converted",
    )


def test_disks_devices_dict(snapshot: SnapshotAssertion):
    """Test disks devices dict."""
    disk_dict = asdict(FIXTURE_DISKS.devices[0])
    assert isinstance(disk_dict, dict)
    assert disk_dict == snapshot

    disk_converted = Disk(**disk_dict)
    assert isinstance(disk_converted, Disk)
    assert disk_converted == snapshot(
        name="disks-devices-dict-converted",
    )


def test_disks_devices_partition_dict(snapshot: SnapshotAssertion):
    """Test disks devices partition dict."""
    partition_dict = asdict(FIXTURE_DISKS.devices[0].partitions[0])
    assert isinstance(partition_dict, dict)
    assert partition_dict == snapshot

    partition_converted = DiskPartition(**partition_dict)
    assert isinstance(partition_converted, DiskPartition)
    assert partition_converted == snapshot(
        name="disks-devices-partition-dict-converted",
    )
