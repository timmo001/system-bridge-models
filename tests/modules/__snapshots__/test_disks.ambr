# serializer version: 1
# name: test_disks
  Disks(devices=[Disk(name='name', partitions=[DiskPartition(device='device', mount_point='mountpoint', filesystem_type='fstype', options='options', max_file_size=1, max_path_length=2, usage=DiskUsage(total=1, used=2, free=3, percent=40.2))], io_counters=DiskIOCounters(read_count=3, write_count=4, read_bytes=1, write_bytes=2, read_time=5, write_time=6))], io_counters=DiskIOCounters(read_count=3, write_count=4, read_bytes=1, write_bytes=2, read_time=5, write_time=6))
# ---
# name: test_disks_devices_dict
  dict({
    'io_counters': dict({
      'read_bytes': 1,
      'read_count': 3,
      'read_time': 5,
      'write_bytes': 2,
      'write_count': 4,
      'write_time': 6,
    }),
    'name': 'name',
    'partitions': list([
      dict({
        'device': 'device',
        'filesystem_type': 'fstype',
        'max_file_size': 1,
        'max_path_length': 2,
        'mount_point': 'mountpoint',
        'options': 'options',
        'usage': dict({
          'free': 3,
          'percent': 40.2,
          'total': 1,
          'used': 2,
        }),
      }),
    ]),
  })
# ---
# name: test_disks_devices_dict[disks-devices-dict-converted]
  Disk(name='name', partitions=[DiskPartition(device='device', mount_point='mountpoint', filesystem_type='fstype', options='options', max_file_size=1, max_path_length=2, usage=DiskUsage(total=1, used=2, free=3, percent=40.2))], io_counters=DiskIOCounters(read_count=3, write_count=4, read_bytes=1, write_bytes=2, read_time=5, write_time=6))
# ---
# name: test_disks_devices_partition_dict
  dict({
    'device': 'device',
    'filesystem_type': 'fstype',
    'max_file_size': 1,
    'max_path_length': 2,
    'mount_point': 'mountpoint',
    'options': 'options',
    'usage': dict({
      'free': 3,
      'percent': 40.2,
      'total': 1,
      'used': 2,
    }),
  })
# ---
# name: test_disks_devices_partition_dict[disks-devices-partition-dict-converted]
  DiskPartition(device='device', mount_point='mountpoint', filesystem_type='fstype', options='options', max_file_size=1, max_path_length=2, usage=DiskUsage(total=1, used=2, free=3, percent=40.2))
# ---
# name: test_disks_dict
  dict({
    'devices': list([
      dict({
        'io_counters': dict({
          'read_bytes': 1,
          'read_count': 3,
          'read_time': 5,
          'write_bytes': 2,
          'write_count': 4,
          'write_time': 6,
        }),
        'name': 'name',
        'partitions': list([
          dict({
            'device': 'device',
            'filesystem_type': 'fstype',
            'max_file_size': 1,
            'max_path_length': 2,
            'mount_point': 'mountpoint',
            'options': 'options',
            'usage': dict({
              'free': 3,
              'percent': 40.2,
              'total': 1,
              'used': 2,
            }),
          }),
        ]),
      }),
    ]),
    'io_counters': dict({
      'read_bytes': 1,
      'read_count': 3,
      'read_time': 5,
      'write_bytes': 2,
      'write_count': 4,
      'write_time': 6,
    }),
  })
# ---
# name: test_disks_dict[disks-dict-converted]
  Disks(devices=[Disk(name='name', partitions=[DiskPartition(device='device', mount_point='mountpoint', filesystem_type='fstype', options='options', max_file_size=1, max_path_length=2, usage=DiskUsage(total=1, used=2, free=3, percent=40.2))], io_counters=DiskIOCounters(read_count=3, write_count=4, read_bytes=1, write_bytes=2, read_time=5, write_time=6))], io_counters=DiskIOCounters(read_count=3, write_count=4, read_bytes=1, write_bytes=2, read_time=5, write_time=6))
# ---
